src/instruction/instructions/jump/blx.ts

```diff
@@ -42,9 +41,8 @@ export class BlxInstruction extends BaseInstruction {
     const rmRegister = getBits(opcode[0], this.pattern).value
     const content = registers.readRegister(rmRegister)
     if (content.toUnsignedInteger() % 2 !== 0) {
-      throw new VirtualBoardError(
-        `value of register cannot be divided by 2 and and is therefore not valid`,
-        VirtualBoardErrorType.InvalidParamProvided
+      throw new Error(
+        'Value of register cannot be divided by 2 and and is therefore not valid.'
```

Comment:
IMO: user has impact what value is in defined register. so this would be a runtime exception for me

Reply:
Yes but the encoder has to make sure that the instructions are aligned and that the passed offset is dividable by two. So I would argue that yes, theoretically the user can have an impact but it would be bad if he could, because otherwise we have the issue that we cannot properly calculate offsets. So if the passed value is not dividable by two there is a bug in the encoder.